blog/content/post/2020-12-14-docker-testing.md

```diff
@@ -0,0 +1,375 @@
+---
+title: Containerization of the test environment for embedded systems
+abstract: 'Using Docker makes life much easier for developers. It allows us to
+          build lightweight and portable software containers that simplify
+          application development. In this article, we present the advantages of
+          using Docker for embedded testing
+          '
+cover: /covers/docker_testing.jpg
+author: mateusz.grzelak
+layout: post
+published: true
+date: 2020-12-14
+archives: "2020"
+
+tags:
+  - docker
+  - containerization
+  - testing
+  - container
+categories:
+  - Firmware
+
+---
+
+# Introduction
+
+Docker is an open-source tool used for creating, deploying, and running 
+applications using containers. It performs operating-system-level
+virtualization, also known as “containerization”. A container is a standard unit
+of code with libraries, dependencies and other configuration and binaries needed
+to run it.
+
+Containers allow applications to run independently of the system they are
+running on. It is extremely useful when creating an application that runs on our
+computer, but not on another programmer's computer. It is possible that a
+program that uses many dependencies and libraries may not run on another
+computer due to incorrect versions. In this case, we only need to define
+necessary dependencies and pack our application in a container and regardless of
+the operating system, it should run on each computer with an installed Docker.
```

Comment:
Containers allow applications to run independently of the system they are running on. It is handy when creating an application that runs on our computer but not on another programmer's computer. A program that uses many dependencies and libraries may not run on another computer due to incorrect versions. In this case, we only need to define necessary dependencies and pack our application in a container. Regardless of the operating system, it should run on each computer with an installed Docker.